### PR TITLE
SNOW-1829928 Fix a bug related to df.loc set

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1829928 Fix a bug related to df.loc set

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

The bug fix is simply make sure the input of columns to internal method should not be a scalar.

### CODE REVIEW

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Thanks for the quick fix!

**Review by sfc-gh-joshi — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,17 +54,14 @@
...
-#### New Features
-
 - Added support for `DataFrame.align` and `Series.align` for `axis=1` and `axis=None`.
 - Added support fot `pd.json_normalize`.
 - Added support for `GroupBy.pct_change` with `axis=0`, `freq=None`, and `limit=None`.
 
+#### Dependency Updates
+
 #### Bug Fixes
+- Fixed a bug in `df.loc` where setting a single column from a series results unexpected `None` values.
```
Comment (sfc-gh-nkumar) [inline CHANGELOG.md:64]: typo: results unexpected -> results in unexpected

**Review by sfc-gh-nkumar — APPROVED** [review]

**Review by sfc-gh-jjiao — APPROVED** [review]

File: tests/integ/modin/frame/test_loc.py [hunk 2]
```diff
@@ -4306,3 +4306,31 @@ def test_df_loc_set_series_value_slice_key(key, row_loc):
             snow_df.loc[row_loc, key] = pd.Series([1, 4, 9], index=list("ABC"))
             native_df.loc[row_loc, key] = native_pd.Series([1, 4, 9], index=list("ABC"))
         assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(snow_df, native_df)
+
+
+@sql_count_checker(query_count=3)
+def test_fix_1829928():
```
[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply tests/integ/modin/frame/test_loc.py:4312]: Oh, the PR is merged already. If you insist, I can fire another PR to rename it.

[the remark this answers is not in the document]
Comment (sfc-gh-jjiao) [reply tests/integ/modin/frame/test_loc.py:4312]: no, that's fine. just some nit picking

**Pull request merged**